collectors/python.d.plugin/spigotmc/spigotmc.chart.py

```diff
@@ -49,6 +50,10 @@ def __init__(self, configuration=None, name=None):
         self.password = self.configuration.get('password', '')
         self.console = mcrcon.MCRcon()
         self.alive = True
+        self._tps_regex = re.compile(
+            b'^.*: ([0-9]{1,2}.[0-9]{1,2}), .*?([0-9]{1,2}.[0-9]{1,2}), .*?([0-9]{1,2}.[0-9]{1,2}).*$'
+        )
+        self._list_regex = re.compile(b'^.*: .*?([0-9]*).*?[0-9]*.*$')
```

Comment:
i suggest to make those regexes global. 

And please use

```
(?P<name>...)
Similar to regular parentheses, but the substring matched by the group is accessible via the symbolic group name name. Group names must be valid Python identifiers, and each group name must be defined only once within a regular expression. A symbolic group is also a numbered group, just as if the group were not named.
```

this is much clear

Reply:
or maybe something like this

https://github.com/netdata/netdata/blob/2ee2d94891e88d9c773f7633746a38fb341f2bac/collectors/python.d.plugin/smartd_log/smartd_log.chart.py#L63-L74

Reply:
it is nice to have comments:

```python
re.compile( 
     '(\d+);'  # attribute 
     '(\d+);'  # normalized value 
     '(\d+)',  # raw value 
     re.X 
 )
```
vs
```python
re.compile('(\d+);(\d+);(\d+)')
```

Reply:
Doh!  Should have though of both points.  I've been too far out of Python since the holidays...